lib/utils/dec.c

```diff
@@ -3,31 +3,49 @@
  *
  * SPDX-License-Identifier: Apache-2.0
  */
-
+#include <limits.h>
+#include <string.h>
 #include <zephyr/sys/util.h>
 
 uint8_t u8_to_dec(char *buf, uint8_t buflen, uint8_t value)
 {
-	uint8_t divisor = 100;
-	uint8_t num_digits = 0;
-	uint8_t digit;
-
-	while ((buflen > 0) && (divisor > 0)) {
-		digit = value / divisor;
-		if ((digit != 0) || (divisor == 1) || (num_digits != 0)) {
-			*buf = digit + (char)'0';
-			buf++;
-			buflen--;
-			num_digits++;
-		}
-
-		value -= digit * divisor;
-		divisor /= 10;
-	}
+	/* Assume base 10 */
+	return (uint8_t)itoa_util(value, buf, buflen, 10);
+}
 
-	if (buflen != 0) {
+int itoa_util(int value, char *buf, size_t buflen, int base)
+{
+	size_t num_digits = 0;
+	char buffer[sizeof(value) * CHAR_BIT + 2];
+	static const char digits[37] = "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ";
+
+	/* Check for valid base */
+	if (base < 2 || base > 36) {
 		*buf = '\0';
 	}
```

Comment:
Two items.

1. If base is out of range, I we ought to return early.
2. There also ought to be a check for buf == NULL.

Reply:
excellent points! Changes have been made